tests/integration/generic_test.go

```diff
@@ -28,75 +27,79 @@ var _ = Describe("odo generic", func() {
 			Expect(output).To(ContainSubstring("To see a full list of commands, run 'odo --help'"))
 		})
+	})
 
-		It("Make sure we have the full description when performing odo --help", func() {
-			output := helper.Cmd("odo", "--help").ShouldPass().Out()
-			Expect(output).To(ContainSubstring("Use \"odo [command] --help\" for more information about a command."))
+	When("using an invalid subcommand", func() {
+		It("an error message and help including Usage, Examples, Available commands, and flags is returned", func() {
+			output := helper.Cmd("odo", "hello").ShouldFail().Err()
+			Expect(output).To(ContainSubstring("Usage:"))
+			Expect(output).To(ContainSubstring("Examples:"))
+			Expect(output).To(ContainSubstring("Commands:"))
+			Expect(output).To(ContainSubstring("Flags:"))
 		})
+	})
 
+	When("trying to create a component with an invalid component name", func() {
 		It("Fail when entering an incorrect name for a component", func() {
 			output := helper.Cmd("odo", "component", "foobar").ShouldFail().Err()
-			Expect(output).To(ContainSubstring("Subcommand not found, use one of the available commands"))
+			Expect(output).To(ContainSubstring("Error: Subcommand not found, use one of the available commands:"))
+			Expect(output).To(ContainSubstring("Usage:"))
+			Expect(output).To(ContainSubstring("Examples:"))
+			Expect(output).To(ContainSubstring("Available Commands:"))
+			Expect(output).To(ContainSubstring("Flags:"))
+			Expect(output).To(ContainSubstring("Additional Flags:"))
 		})
-
-		It("Fail with showing help only once for incorrect command", func() {
-			output := helper.Cmd("odo", "hello").ShouldFail().Err()
-			Expect(strings.Count(output, "odo [flags]")).Should(Equal(1))
-		})
-
 	})
 
-	Context("When executing catalog list without component directory", func() {
-		It("should list all component catalogs", func() {
+	When("executing catalog list without component directory", func() {
+		It("should list all devfile components", func() {
 			stdOut := helper.Cmd("odo", "catalog", "list", "components").ShouldPass().Out()
-			helper.MatchAllInOutput(stdOut, []string{"nodejs", "python", "php", "go", "java"})
+			helper.MatchAllInOutput(stdOut, []string{"Devfile", "nodejs", "python", "php", "go", "java"})
 		})
-
 	})
 
-	Context("check catalog component search functionality", func() {
-		It("check that a component does not exist", func() {
+	When("searching the catalog for a non existing component", func() {
+		It("searches for the component and returs message indicating that no component matched the query", func() {
 			componentRandomName := helper.RandString(7)
 			output := helper.Cmd("odo", "catalog", "search", "component", componentRandomName).ShouldFail().Err()
 			Expect(output).To(ContainSubstring("no component matched the query: " + componentRandomName))
 		})
 	})
 
 	// Test machine readable output
-	Context("when creating project -o json", func() {
+	When("creating an application using -o json flag", func() {
```

Comment:
This should be kept as it is. This test is creating a project. See https://github.com/redhat-developer/odo/pull/5406/files#diff-bb9e5f2fbc1b8edb31629193e23d6ec67b8963e0b730d9cb1f89305cd561a266R97